src/ucp/wireup/wireup_cm.c

```diff
@@ -0,0 +1,272 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2019.  ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include "wireup_cm.h"
+#include <ucp/core/ucp_ep.inl>
+#include <ucp/wireup/wireup.h>
+#include <ucp/wireup/wireup_ep.h>
+
+
+static unsigned
+ucp_cm_ep_init_flags(const ucp_worker_h worker, const ucp_ep_params_t *params)
+{
+    if (!ucp_worker_sockaddr_is_cm_proto(worker)) {
```

Comment:
really need to check this? maybe ucs_assert?

Reply:
this will be used from common code in next PRs, for example here
https://github.com/evgeny-leksikov/ucx/blob/07507c6600a35ebae1c458063fa68a5108ba00ac/src/ucp/wireup/wireup.c#L646

Reply:
seems like error flow

Reply:
why? the func just connects lane of any type